hooks/useSavedGuideIds.tsx

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from "react";
+import { SAVED_GUIDES_LOCAL_STORAGE } from "utils/constants";
+
+function getGuideIds() {
+  const rawSavedGuideIds = localStorage.getItem(SAVED_GUIDES_LOCAL_STORAGE);
+
+  const savedGuideIds: string[] = (() => {
+    try {
+      return JSON.parse(rawSavedGuideIds || "[]") || [];
+    } catch (err) {
+      console.error(
+        "Problem parsing shared guides, returning empty array...",
+        err
+      );
+      return [];
+    }
+  })();
+
+  return savedGuideIds;
+}
+
+export function useSavedGuideIds() {
+  const [savedGuideIds, setSavedGuideIds] = useState<string[]>(getGuideIds());
+
+  const addSavedGuideId = (newId: string) => {
+    if (savedGuideIds.indexOf(newId) !== -1) {
+      setSavedGuideIds([...savedGuideIds, newId]);
+    }
+  };
+
+  useEffect(() => {
+    if (savedGuideIds.length > 0) {
+      const joinedIds = JSON.stringify(savedGuideIds);
+      localStorage.setItem(SAVED_GUIDES_LOCAL_STORAGE, joinedIds);
+    }
+  }, [savedGuideIds, setSavedGuideIds]);
```

Comment:
`setSavedGuideIds` is not required as a dependency here.

Reply:
great catch! I think it was a dependency at some point and I forgot to remove it